### DISCUSSION TITLE
document.delete_object(xref) is giving an Attribute error

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#1478] [Looking for help]
![image](https://user-images.githubusercontent.com/36301306/146901685-c89cc38d-a04f-450b-a802-a183646f13f6.png)

**Comment by JorjMcKie** [comment]
This method no longer exists - the documentation needs this update.

**Reply by meghanaviyyapu** [reply]
Okay Thanks.

**Reply by marcfalk** [reply]
> This method no longer exists - the documentation needs this update.

Is there an alternative to doc.delete_object? Say if I wanted to delete an image or a piece of text based on its xref.

**Reply by JorjMcKie** [reply]
This method could never be used for the purposes you mentioned.
1. Text is not linked to a specific xref, so the method would not be relevant here anyway.
2. Image deletion also can never achieved by simply removing its xref.

To remove text use redaction annotations. To remove an image follow the guideline given [here](https://github.com/pymupdf/PyMuPDF-Utilities/tree/master/image-replacement).

`doc.delete_xref()` always was too dangerous because of the network of crosswise dependencies of objects in a PDF.
